--- frameworks/av/media/libstagefright/colorconversion/SoftwareRenderer.cpp
+++ frameworks/av/media/libstagefright/colorconversion/SoftwareRenderer.cpp
@@ -126,6 +126,12 @@
     {
         switch (mColorFormat) {
             case OMX_COLOR_FormatYUV420Planar:
+            {
+                halFormat = HAL_PIXEL_FORMAT_RGBA_8888;
+                bufWidth = (mCropWidth + 1) & ~1;
+                bufHeight = (mCropHeight + 1) & ~1;
+                break;
+            }
             case OMX_COLOR_FormatYUV420SemiPlanar:
             case OMX_TI_COLOR_FormatYUV420PackedSemiPlanar:
             {
@@ -180,6 +186,11 @@
         mConverter = new ColorConverter(
                 mColorFormat, OMX_COLOR_FormatYUV444Y410);
         CHECK(mConverter->isValid());
+    } else if (halFormat == HAL_PIXEL_FORMAT_RGBA_8888 &&
+        mColorFormat == OMX_COLOR_FormatYUV420Planar) {
+        mConverter = new ColorConverter(
+                mColorFormat, OMX_COLOR_Format32BitRGBA8888);
+        CHECK(mConverter->isValid());
     }
 
     CHECK(mNativeWindow != NULL);
--- frameworks/av/media/codec2/vndk/C2Store.cpp
+++ frameworks/av/media/codec2/vndk/C2Store.cpp
@@ -850,7 +850,7 @@
     emplace("libcodec2_soft_amrwbenc.so");
     //emplace("libcodec2_soft_av1dec_aom.so"); // deprecated for the gav1 implementation
     emplace("libcodec2_soft_av1dec_gav1.so");
+    //emplace("libcodec2_soft_avcdec.so");
-    emplace("libcodec2_soft_avcdec.so");
     emplace("libcodec2_soft_avcenc.so");
     emplace("libcodec2_soft_flacdec.so");
     emplace("libcodec2_soft_flacenc.so");
@@ -871,7 +871,7 @@
     emplace("libcodec2_soft_vorbisdec.so");
     emplace("libcodec2_soft_vp8dec.so");
     emplace("libcodec2_soft_vp8enc.so");
+    //emplace("libcodec2_soft_vp9dec.so");
-    emplace("libcodec2_soft_vp9dec.so");
     emplace("libcodec2_soft_vp9enc.so");
 }
 
--- hardware/interfaces/bluetooth/1.0/default/vendor_interface.cc
+++ hardware/interfaces/bluetooth/1.0/default/vendor_interface.cc
@@ -342,7 +342,7 @@
   lib_interface_->op(BT_VND_OP_GET_LPM_IDLE_TIMEOUT, &lpm_timeout_ms);
   ALOGI("%s: lpm_timeout_ms %d", __func__, lpm_timeout_ms);
 
-  bt_vendor_lpm_mode_t mode = BT_VND_LPM_ENABLE;
+  bt_vendor_lpm_mode_t mode = BT_VND_LPM_DISABLE;
   lib_interface_->op(BT_VND_OP_LPM_SET_MODE, &mode);
 
   ALOGD("%s Calling StartLowPowerWatchdog()", __func__);
--- packages/apps/TvSettings/Settings/res/layout/setup_password_item.xml
+++ packages/apps/TvSettings/Settings/res/layout/setup_password_item.xml
@@ -32,7 +32,7 @@
         android:paddingEnd="@dimen/setup_list_item_padding"
         android:paddingStart="@dimen/setup_list_item_padding">
 
-        <EditText
+        <androidx.leanback.widget.GuidedActionEditText
             android:id="@+id/guidedactions_item_title"
             style="@style/Setup.Action.TextInput"
             android:layout_width="match_parent"
@@ -40,8 +40,8 @@
             android:focusable="true"
             android:gravity="center_vertical"
             android:imeOptions="actionNext|flagNoExtractUi"
-            android:inputType="text">
-        </EditText>
+            android:inputType="text" />
+
     </FrameLayout>
 
     <CheckBox
